### PR TITLE
fix: update token balances query provided tags

### PR DESCRIPTION
**Pull request by josheleonard**
Fine-tunes the cache for querying multiple balances

### CODE REVIEW

File: components/brave_wallet_ui/common/slices/api.slice.ts [hunk 1]
```diff
@@ -1483,15 +1494,14 @@ export function createWalletApi (
...
-                {
+            ? ['TokenBalancesForChainId', 'UNKNOWN_ERROR']
+            : (balancesResult &&
+                Object.keys(balancesResult).map((tokenAddress) => ({
                   type: 'TokenBalancesForChainId',
-                  id: `${arg.address}-${arg.coin}-${arg.chainId}`
-                }
-              ]
+                  id: `${arg.address}-${arg.coin}-${arg.chainId}-${tokenAddress}`
+                }))) || ['TokenBalancesForChainId']
```
Comment (onyb) [inline components/brave_wallet_ui/common/slices/api.slice.ts:1504]: Then refactor here based on the other suggestion.

File: components/brave_wallet_ui/page/screens/swap/swap.tsx [hunk 2]
```diff
@@ -93,13 +98,21 @@ export const Swap = () => {
...
+          ? {
+              contracts,
+              address,
+              coin,
+              chainId
+            }
+          : {
+              address,
+              coin,
+              chainId
```
Comment (onyb) [inline components/brave_wallet_ui/page/screens/swap/swap.tsx:113]: Prefer throwing an exception for anything other than ETH, and add support in a future PR. In any case, it'll need changes after the Swap migration PR is merged.

**Review by onyb — APPROVED** [review]
Looks good overall. I left some small remarks to make things more readable, and avoid having to treat evm and solana using the same semantics of "address" and "contracts".

File: components/brave_wallet_ui/common/slices/api.slice.ts (hunk 1 above)
Comment (josheleonard) [reply components/brave_wallet_ui/common/slices/api.slice.ts]: refactored

File: components/brave_wallet_ui/page/screens/swap/swap.tsx (hunk 2 above)
Comment (josheleonard) [reply components/brave_wallet_ui/page/screens/swap/swap.tsx]: now throwing an exception for anything other than ETH

**Pull request merged**